libsolidity/analysis/FunctionCallGraph.cpp

```diff
@@ -0,0 +1,246 @@
+			if (!m_graph->edges.count(funcDef))
+				visitCallable(funcDef);
+	}
+
+	// Add all CreationDispatch calls to the RuntimeDispatch as well
+	for (auto node: m_graph->edges[SpecialNode::CreationDispatch])
+		add(SpecialNode::RuntimeDispatch, node);
+
+	// Add all external functions to the RuntimeDispatch
+	for (auto& [hash, functionType]: _contract.interfaceFunctionList())
+	{
+		(void)hash;
+		add(SpecialNode::RuntimeDispatch, &functionType->declaration());
+	}
+
+	if (_contract.fallbackFunction())
+		add(SpecialNode::RuntimeDispatch, _contract.fallbackFunction());
+
+	if (_contract.receiveFunction())
+		add(SpecialNode::RuntimeDispatch, _contract.receiveFunction());
+
+	m_contract = nullptr;
+	solAssert(!nodeSet(m_currentNode), "Current node not properly reset.");
+
+	return m_graph;
+}
+
+bool FunctionCallGraphBuilder::visit(Identifier const& _identifier)
+{
+	if (auto const* callable = dynamic_cast<CallableDeclaration const*>(_identifier.annotation().referencedDeclaration))
+	{
+		solAssert(*_identifier.annotation().requiredLookup == VirtualLookup::Virtual, "");
+
+		auto funType = dynamic_cast<FunctionType const*>(_identifier.annotation().type);
+
+		// For events kind() == Event
+		if (funType && funType->kind() == FunctionType::Kind::Internal)
+		{
+			processFunction(callable->resolveVirtual(*m_contract), _identifier.annotation());
+
+			solAssert(nodeSet(m_currentNode), "");
+		}
+	}
+
+	return true;
+}
+
+bool FunctionCallGraphBuilder::visit(NewExpression const& _newExpression)
+{
+	if (ContractType const* contractType = dynamic_cast<ContractType const*>(_newExpression.typeName().annotation().type))
+		m_graph->createdContracts.emplace(&contractType->contractDefinition());
+
+	return true;
+}
+
+bool FunctionCallGraphBuilder::visit(MemberAccess const& _memberAccess)
+{
+	// Bound functions
+	if (auto funType = dynamic_cast<FunctionType const*>(_memberAccess.annotation().type))
+		if (funType->bound() && funType->kind() == FunctionType::Kind::Internal)
+		{
+			processFunction(dynamic_cast<CallableDeclaration const&>(funType->declaration()), _memberAccess.annotation());
+			return true;
+		}
+
+	// Direct access functions like C.foo()
+	if (TypeType const* type = dynamic_cast<TypeType const*>(_memberAccess.expression().annotation().type))
+
+		if (dynamic_cast<ContractType const*>(type->actualType()))
+			if (auto funType = dynamic_cast<FunctionType const*>(_memberAccess.annotation().type))
+				if (funType->kind() == FunctionType::Kind::Internal)
+					if (auto const* function = dynamic_cast<FunctionDefinition const*>(_memberAccess.annotation().referencedDeclaration))
+					{
+						solAssert(*_memberAccess.annotation().requiredLookup == VirtualLookup::Static, "");
+						processFunction(*function, _memberAccess.annotation());
+						return true;
+					}
+
+	// Free functions referenced through modules
+	if (_memberAccess.expression().annotation().type->category() == Type::Category::Module)
```

Comment:
Actually can you put this at the end as the "else" clause? It's basically an internal function call referenced in whatever way.